### PR TITLE
Add CampaignsFolder and new functions in Campaigns for mailchimp__marketing

### PR DESCRIPTION
**Pull request by u3kkasha**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mailchimp.com/developer/marketing/api

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/mailchimp__mailchimp_marketing/mailchimp__mailchimp_marketing-tests.ts
1. types/mailchimp__mailchimp_marketing/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7c4d40c073a3e3b54f9d23cf83f333a8b32eedeb
</p>

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 1]
```diff
@@ -2542,6 +2652,492 @@ export namespace campaigns {
...
+                    /**
+                     * Sends the daily RSS Campaign on Saturdays.
+                     */
+                    saturday?: boolean;
+                };
+
+                /**
+                 * The day of the week to send a weekly RSS Campaign. Possible values: "sunday", "monday", "tuesday", "wednesday", "thursday", "friday", or "saturday".
+                 */
+                weekly_send_day?: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2719]:
```suggestion
                weekly_send_day?: "sunday" | "monday" | "tuesday" | "wednesday" | "thursday" | "friday" | "saturday";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 2]
```diff
@@ -2542,6 +2652,492 @@ export namespace campaigns {
...
+            segment_opts?: {
+                /**
+                 * The id for an existing saved segment.
+                 */
+                saved_segment_id?: number;
+
+                /**
+                 * Segment match type. Possible values: "any" or "all".
+                 */
+                match?: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2762]:
```suggestion
                match?: "any" | "all";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 3]
```diff
@@ -2542,6 +2652,492 @@ export namespace campaigns {
...
+         */
+        url?: string;
+        }>;
+    }
+
+    interface ChecklistItem {
+        /**
+         * The item type. Possible values: "success", "warning", or "error".
+         */
+        type: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:3098]:
```suggestion
        type: "success" | "warning" | "error";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 4]
```diff
@@ -2542,6 +2648,476 @@ export namespace campaigns {
...
+
+            /**
+             * The URL for the RSS feed.
+             */
+            feed_url: string;
+
+            /**
+             * The frequency of the RSS Campaign. Possible values: "daily", "weekly", or "monthly".
+             */
+            frequency: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2734]:
```suggestion
            frequency: "daily" | "monthly" | "weekly";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 5]
```diff
@@ -2542,6 +2648,476 @@ export namespace campaigns {
...
+
+            /**
+             * The possible reply-to addresses. The number of reply_to_addresses provided must match the number of from_names. If no reply_to_addresses are provided, settings.reply_to will be used.
+             */
+            reply_to_addresses?: string[];
+
+            /**
+             * The combination that performs the best. This may be determined automatically by click rate, open rate, or total revenue -- or you may choose manually based on the reporting data you find the most valuable. For Multivariate Campaigns testing send_time, winner_criteria is ignored. For Multivariate Campaigns with 'manual' as the winner_criteria, the winner must be chosen in the Mailchimp web application. Possible values: "opens", "clicks", "manual", or "total_revenue".
+             */
+            winner_criteria: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2798]:
```suggestion
            winner_criteria: "opens" | "clicks" | "manual" | "total_revenue";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 6]
```diff
@@ -2542,6 +2648,476 @@ export namespace campaigns {
...
+                /**
+                 * The title for the card. Typically the subject line of the campaign.
+                 */
+                title?: string;
+            };
+
+            /**
+             * How the campaign's content is put together. The old drag and drop editor uses 'template' while the new editor uses 'multichannel'. Defaults to template. Possible values: "template" or "multichannel".
+             */
+            content_type?: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2968]:
```suggestion
            content_type?: "template" | "multichannel";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 7]
```diff
@@ -2542,6 +2648,476 @@ export namespace campaigns {
...
+    interface SetContentParams {
+        /**
+         * Available when uploading an archive to create campaign content. The archive should include all campaign content and images.
+         * [Learn more](https://mailchimp.com/help/import-a-custom-html-template/)
+         */
+        archive?: {
+            /**
+             * The type of encoded file. Defaults to zip. Possible values: "zip", "tar.gz", "tar.bz2", "tar", "tgz", or "tbz".
+             */
+            archive_type?: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:2981]:
```suggestion
            archive_type?: "zip" | "tar.gz" | "tar.bz2" | "tar" | "tgz" | "tbz";
```

File: types/mailchimp__mailchimp_marketing/index.d.ts [hunk 8]
```diff
@@ -2542,6 +2648,476 @@ export namespace campaigns {
...
+        variate_contents?: Array<{
+            /**
+             * Available when uploading an archive to create campaign content. The archive should include all campaign content and images.
+             * [Learn more](https://mailchimp.com/help/import-a-custom-html-template/)
+             */
+            archive?: {
+                /**
+                 * The type of encoded file. Defaults to zip. Possible values: "zip", "tar.gz", "tar.bz2", "tar", "tgz", or "tbz".
+                 */
+                archive_type?: string;
```
Comment (codershiba) [inline types/mailchimp__mailchimp_marketing/index.d.ts:3031]:
```suggestion
                archive_type?: "zip" | "tar.gz" | "tar.bz2" | "tar" | "tgz" | "tbz";
```

**Review by ghost — APPROVED** [review]

**Comment by u3kkasha** [comment]
Yeah,
'Approve Request'

**Pull request merged**